back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -250,6 +277,20 @@ def update
     CustomFieldService.new.compact_custom_field_values! update_params[:custom_field_values]
     input.set_manual_votes(update_params[:manual_votes_amount], current_user) if update_params[:manual_votes_amount]
 
+    creation_phase = input.creation_phase
+
+    if (input.publication_status == 'published' || update_params[:publication_status] == 'published') && UserFieldsInSurveyService.should_merge_user_fields_into_idea?(
```

Comment:
Not sure if the `input.publication_status` part here is necessary, since I don't think regular people submitting a survey can ever update an already published survey